imageio/core/imopen.pyi

```diff
@@ -25,6 +26,16 @@ def imopen(
     **kwargs
 ) -> PillowPlugin: ...
 @overload
+def imopen(
+    uri,
+    io_mode: Literal["r", "w"],
+    *,
+    plugin: Literal["opencv"],
+    format_hint: str = None,
+    legacy_mode: Literal[False],
+    **kwargs
+) -> OpenCVPlugin: ...
```

Comment:
This is what I meant - why are there multiple imopen versions here, for a few specific formats?

Reply:
This is typing overload for different plugins. In modern IDEs this will give you indicators like the one in this screenshot. Notice the 1/4 that you can also find in built-ins, numpy functions or similar.

![image](https://user-images.githubusercontent.com/4402489/163979441-9ba7c243-bc31-442e-b5cd-e5e20ffcf19c.png)

(Sidenote: looking at the picture, we may wish to reorder the overloads so it starts with hinting the generic one though; I thought it would start with the last one.)

The other benefit is that validators like mypy or intelli-sense pick up on the overloads for different literals, which means that they can now statically infer that `iio.imopen(..., plugin="pyav")` will return a PyAV plugin. As a result, you will get plugin-specific autocompletion and documentation for the respective plugin, which should yield better discoverability/UX and will allow mypy to validate that plugin-specific kwargs are used correctly.

![image](https://user-images.githubusercontent.com/4402489/163981259-d801c7f8-eea3-4a6f-83f2-2487347221af.png)

Reply:
Ok, and the reason why these 3 plugins have their own type-signature and not others because these deviate more from the generic one?

Reply:
@almarklein Yes, basically. Each subclass of `PluginV3` that we have in the library will get a dedicated entry here so that users can discover and validate any special methods and kwargs that these plugins offer.